protocols/mdns/src/behaviour.rs

```diff
@@ -107,6 +164,77 @@ impl Mdns {
     pub fn discovered_nodes(&self) -> impl ExactSizeIterator<Item = &PeerId> {
         self.discovered_nodes.iter().map(|(p, _, _)| p)
     }
+
+    fn inject_mdns_packet(&mut self, packet: MdnsPacket, params: &impl PollParameters) {
+        self.timeout.set_interval(self.interval);
```

Comment:
Say there is one very noisy node on the network broadcasting an mdns packet every 1 minute. With a timeout of e.g. 5 minutes configured locally, the timeout would never fire, thus the local node would never broadcast a query and thus addresses from other nodes would expire locally, correct?

In case I am not mistaken with the assumption above, I would see two ways forward:

- Do not reset the timeout, always sending out a query at each interval.
- Remove the notion of TTLs for address as suggested by David.

I have yet to put more thoughts into this, thus please feel free to ignore the comment.

Reply:
This isn't quite accurate. If a peer sends a query, all other peers respond with their addresses to the multicast address. So you really only need one peer to make the query, as everyone will get the updates from everyone. This means that they will not expire locally.

Reply:
I would prefer to not reset the timeout for now and leave the removal of the TTL to another PR.

Reply:
if the timeout is smaller than the ttl, it should still be safe to reset the timeout. as I said it uses multicast so everyone gets all queries and all responses.

Reply:
Thanks for clarifying @dvc94ch!